pageserver/src/disk_usage_eviction_task.rs

```diff
@@ -529,6 +547,44 @@ pub(crate) async fn disk_usage_eviction_task_iteration_impl<U: Usage>(
         (usage_assumed, evictions_failed)
     };
 
+    let started_at = std::time::Instant::now();
+
+    let evict_layers = async move {
+        let mut evict_layers = std::pin::pin!(evict_layers);
+
+        let maximum_expected = std::time::Duration::from_secs(10);
+        let nag_every = std::time::Duration::from_secs(2);
+
+        match tokio::time::timeout(maximum_expected, &mut evict_layers).await {
+            Ok(tuple) => return tuple,
+            Err(_timeout) => {
+                tracing::info!(
+                    elapsed_ms = started_at.elapsed().as_millis(),
+                    "still ongoing"
+                );
+            }
+        };
+
+        loop {
+            let res = tokio::time::timeout(nag_every, &mut evict_layers).await;
+
+            let elapsed_ms = started_at.elapsed().as_millis();
+
+            match res {
+                Ok(tuple) => {
+                    tracing::info!(%elapsed_ms, "completed");
+                    return tuple;
+                }
+                Err(_timeout) => {
+                    tracing::info!(%elapsed_ms, "still ongoing");
+                }
+            }
+        }
```

Comment:
This could be DRY with

```rust
for i in 0.. {
    match tokio::time::timeout(maximum_expected, &mut evict_layers).await {
        Ok(tuple) => {
            if i > 0 {
                // make noise only if we logged "still going" earlier"
                info!("completed");
            }
            return tuple;   
        }
        Err(_timeout) => {
            tracing::info!(
                elapsed_ms = started_at.elapsed().as_millis(),
                "still ongoing"
            );
            continue;
        }
    }

}
```

Reply:
I removed the repeated nagging after first writing out this PR I was annoyed seeing those in prod in other context. Now there's one nag, then final completion report.